be/src/storage/lake/gc.cpp

```diff
@@ -99,7 +101,68 @@ static Status write_orphan_list_file(const std::set<std::string>& orphans, Writa
     return file->close();
 }
 
-static Status delete_tablet_metadata(std::string_view root_location, const std::set<int64_t>& owned_tablets) {
+static StatusOr<TabletMetadataPtrPair> get_meta_pair(TabletManager* tablet_mgr,
+                                                     const std::string& metadata_root_location, const int64_t tablet_id,
+                                                     const int64_t version, const int64_t next_version) {
+    auto path = join_path(metadata_root_location, tablet_metadata_filename(tablet_id, version));
+    auto next_path = join_path(metadata_root_location, tablet_metadata_filename(tablet_id, next_version));
+    auto meta = tablet_mgr->get_tablet_metadata(path, false);
+    if (!meta.ok()) {
+        return meta.status();
+    }
+    auto next_meta = tablet_mgr->get_tablet_metadata(next_path, true);
+    if (!next_meta.ok()) {
+        return next_meta.status();
+    }
+    return std::make_pair(std::move(*meta), std::move(*next_meta));
+}
+
+// free delvec cache by deleted segment files
+static void calc_free_delvec_cache(TabletMetadataPtrPair meta_pair, std::vector<TabletSegmentId>& to_del_tsids) {
```

Comment:
```suggestion
static void calc_free_delvec_cache(TabletMetadataPtrPair meta_pair, std::vector<TabletSegmentId>* to_del_tsids) {
```
https://starrocks.feishu.cn/docs/doccnH0eI3LRYUa40tsEZjviI3c#vXvR8n